### DISCUSSION TITLE
Web to print error pimcore 10

### DISCUSSION THREAD

**Discussion by mcg-pimcore** [pimcore/pimcore#13198] [Q&A / Support]
Hello,

I am trying to generate pdf using webtoprint feature of pimcore. but when I go to webtoprint setting and test that setting from test button, it is giving me following error:-

wkhtmltopdf reported error (127):

command was: wkhtmltopdf -O landscape '/html/liberty/var/tmp/web2print_6328a5334fbc3.htm' '/html/liberty/var/tmp/web2print_6328a5334fc6b.pdf'

![Screenshot from 2022-09-19 22-52-15](https://user-images.githubusercontent.com/87263262/191078948-d381ad28-4ecd-47b1-a4ab-600f89b40bf6.png)

And got following error When I create print page:-
The controller for URI "/test-print-page" is not callable: Controller "App\Controller\Web2printController" does neither exist as service nor as class.

![Screenshot from 2022-09-19 22-46-20](https://user-images.githubusercontent.com/87263262/191078954-0834da58-0288-481d-b02c-4841c19e9bf1.png)

Please suggest if any setting or installation required for web to print.

**Comment by kingjia90** [comment]
Hi, 

the web2printController is this file
https://github.com/pimcore/skeleton/blob/10.2/src/Controller/Web2printController.php
https://github.com/pimcore/demo/blob/10.2/src/Controller/Web2printController.php
as mentioned in the last phrase [here](https://github.com/pimcore/pimcore/blob/38419b06c61565e62122e32684e37c102cbe25d2/doc/Development_Documentation/03_Documents/02_Document_Types/15_Print_Documents.md#printcontainer) about default implementations .
Please verify that this file is existing in your project root `src` folder and adapt it accordingly to your needs.


The `error 127` is when binary is not found, so wkhtmltopdf is not properly installed, make sure to install, set the proper execution linux permissions and run in the terminal something like `wkhtmltopdf http://pimcore.com pimcore.pdf` to see if it works.

### but before that

maybe it would be useful and good to know that in Pimcore 11, the wkhtmltopdf is not supported anymore (deprecated and [removed](https://github.com/pimcore/pimcore/pull/12842) because it's not officially maintained anymore) and at the moment, afaik [spiritix php chrome html2pdf](https://github.com/spiritix/php-chrome-html2pdf) may be recommended (one potential "drawback" is that it requires NodeJS), while we are working on [adding support for Gotenberg ](https://github.com/pimcore/pimcore/issues/12607)
Hope it helps

**Comment by mcg-pimcore** [comment]
Hi JiaJia,

Thanks for reply.
I will try your answer. Your suggestion to go for chrome html2pdf is good because we are in initial stage of development.
Also no need to worry about nodejs, because we have already installed on system as we are also working on portal engine front-end customisation.
But problem is that there is not proper documentation or tutorial to implement feature in academy. So it is hard to implement. 

So it will be helpful if you can you provide detail to implement 
https://pimcore.com/docs/pimcore/current/Development_Documentation/Documents/Document_Types/Print_Documents.html#page_Settings-for-Headless-Chrome
CC :- @fashxp

**Reply by kingjia90** [reply]
You're welcome.

To install spiritix bundle, it just needs nodejs and `composer require spiritix/php-chrome-html2pdf` and eventually add
```
  "config": {
    "allow-plugins": {
      "eloquent/composer-npm-bridge": true
    }
  }
``` 
to the project root composer (to override the `false` in the pimcore [composer](https://github.com/pimcore/pimcore/blob/281aa3b8245ead539baba1d536876e7f98ba4d85/composer.json#L222))

Maybe this open PR could help https://github.com/pimcore/demo/pull/329


After that, going to `Web-to-print settings` and change the default Tool and it should work. 

My recommandation is to take a look to the demo examples under `print` folder
![image](https://user-images.githubusercontent.com/6014195/191210002-dc3bc7f1-703d-4764-bd49-e359bfe17122.png)

**Reply by mcg-pimcore** [reply]
Hi JiaJia,

Thanks again for reply, I will try this.
